app/scripts/modules/plugins/src/plugins.ts

```diff
@@ -0,0 +1,13 @@
+import { PipelineRegistry } from '@spinnaker/core';
+
+export type IPluginInitialize = (registry: IStageRegistry) => void;
+
+export interface IStageRegistry {
+  pipeline: PipelineRegistry;
+}
+
+declare global {
+  interface Window {
+    spinnakerSettings: any;
```

Comment:
There's a `ISpinnakerSettings` interface defined in `app/scripts/modules/core/src/config/settings.ts`. Better to use that instead of `any`.

Reply:
Wouldn't that mean that a plugin developer would have to import `@spinnaker/core` in order to use `window.spinnakerSettings.onPluginLoaded(plugin);`?

Reply:
I don't think so? Why would they? Is the goal for `@spinnaker/plugin` to not have a dependency on `@spinnaker/core`?

Reply:
Might be worth trying, at least.

Reply:
Okay, I now see what you are saying. Coffee has kicked in!

Reply:
`spinnakerSettings` is actually defined as `ISpinnakerSettings = (window as any).spinnakerSettings` so it has to be of type `any`. We get error TS2717 otherwise.